src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+
+            String[] tokens = input.split(" ", 2);
+
+            String command = tokens[0];
+            String argument = tokens.length == 2 ? tokens[1] : null;
+
+            if (command.equals("bye")) {
+                isExit = executeByeCommand();
+
+            } else if (command.equals("list")) {
+                isExit = executeListCommand();
+
+            } else if(command.equals("done")) {
+                isExit = executeDoneCommand(argument);
+
+            } else if (command.equals("todo")) {
+                isExit = executeTodoCommand(argument);
+
+            } else if (command.equals("deadline")) {
+                isExit = executeDeadlineCommand(argument);
+
+            } else if (command.equals("event")) {
+                isExit = executeEventCommand(argument);
+
+            } else if (command.equals("delete")) {
+                isExit = executeDeleteCommand(argument);
+
+            } else if (command.equals("find")) {
+                isExit = executeFindCommand(argument);
+
+            } else {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+
+            if (isExit) {
+                break;
+            }
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        new Duke().run();
+
+    }
+
+    /**
+     * Executes Bye Command
+     *
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeByeCommand() {
+        System.out.println("Bye. Hope to see you again soon!");
+        storage.save(taskList);
+        return true;
+    }
+
+    /**
+     * Lists out all tasks
+     *
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeListCommand() {
+        if (taskList.size() == 0) {
+            System.out.println("There are no tasks in your task list");
+        } else {
+            System.out.println("Here are the tasks in your task list:");
+            for (int i = 0; i < taskList.size() ; i++ ) {
+                System.out.println((i + 1) + ". "
+                        + taskList.get(i).getTypeIcon()
+                        + taskList.get(i).getStatusIcon() + " "
+                        + taskList.get(i).getDescription()
+                );
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Executes Done Command
+     *
+     * @param argument  command arguments
+     * @return          boolean to represent termination of the program
+     */
+    private boolean executeDoneCommand(String argument) {
+        // check for correct number of arguments
+        if (argument == null) {
```

Comment:
I've noticed these lines being repeated in a few of the executeXXXXXCommand methods. Perhaps it would be better to abstract them out to make the overall code cleaner.

Reply:
Thank you for the suggestion!